### DISCUSSION TITLE
Task image_resource tag value without single quote could be misinterpreted

### DISCUSSION THREAD

**Discussion by tremblaysimon** [concourse/concourse#6997] [Help & Support] [answered]
When I try to use a Docker image with a tag in that format `x.0` where x is a number, the `.0` part seems to be always wiped. Maybe it's something about yaml that I don't properly understand, but when there is a value without single quote I think it can be implicitly converted as a integer or float value. 

Let's look at an example:

```
---
platform: linux
image_resource:
  type: docker-image
  source:
    repository: mcr.microsoft.com/dotnet/sdk
    tag: 5.0
```

Here the tag `5.0` seems to be converted as `5` and since the tag `5` doesn't exist for dotnet/sdk it returns an error when trying to use that task in a pipeline: 

```
resource script '/opt/resource/check []' failed: exit status 1
stderr:
failed to fetch digest for image 'dotnet/sdk:5': 401 Unauthorized
does the image exist?
```

So to be able to use it properly I only add single quote to my tag value like this:

```
    tag: '5.0'
```

Is it a bug in docker-image resource (same behavior in registry-image) or in Concourse or it's just work as expected since it's yaml? In the end, I just want to know if I should always use single quote when I'm using number or I should be able to use a value like 5.0 without single quoting it? What is the best practice if it exists?

**Comment by taylorsilva** [comment, marked as answer]
This is a YAML thing. Best to wrap anything that's not meant to be used as a number (int or float) in double quotes.
https://stackoverflow.com/questions/19109912/yaml-do-i-need-quotes-for-strings-in-yaml

I read a really good blog post about this a few weeks ago but can' find it now :(
